src/mesh/BoundingBox.hpp

```diff
@@ -95,11 +108,18 @@ class BoundingBox {
 private:
   static logging::Logger _log;
 
-  /// Number of dimensions
+  /// Number of dimensions (2 or 3)
   int _dimensions;
 
-  /// Container of min and max points in each dimension
-  std::vector<double> _bounds;
+  /// Container of a Eigen::VectorXd of a bounding box with min components in each dimension
+  /// (x_min, y_min) when _dimensions=2;
+  /// (x_min, y_min, z_min) when _dimensions=3;
+  Eigen::VectorXd _boundMin;
+
+  /// Container of a Eigen::VectorXd of a bounding box with max components in each dimension
+  /// (x_max, y_max) when _dimensions=2;
+  /// (x_max, y_max, z_max) when _dimensions=3;
+  Eigen::VectorXd _boundMax;
```

Comment:
I guess here the important part is not that this is a container, but that it is a point/vertex. And we now have two points, instead of a strange `std::vector`.

Maybe (not sure about `Point A` and `Point B`):

```suggestion
  /// Point A defining the bounding box, with minimum coordinates in each direction
  /// (x_min, y_min) when _dimensions=2;
  /// (x_min, y_min, z_min) when _dimensions=3;
  Eigen::VectorXd _boundMin;

  /// Point B defining the bounding box, with maximum coordinates in each direction
  /// (x_max, y_max) when _dimensions=2;
  /// (x_max, y_max, z_max) when _dimensions=3;
  Eigen::VectorXd _boundMax;
```

Reply:
We change the comment to the following:
``` 
/// _boundMin defining the bounding box, with minimum coordinates in each direction
...
/// _boundMax defining the bounding box, with minimum coordinates in each direction
```

We use _boundMin and _boundMax directly describe the properties of the variables.